testsuite/tests/mgc/test_external_ca.py

```diff
@@ -0,0 +1,53 @@
+"""
+This module contains the most basic happy path test for both DNSPolicy and TLSPolicy
+for a cluster with Let's Encrypt ClusterIssuer
+
+Prerequisites:
+* multi-cluster-gateways ns is created and set as openshift["project"]
+* managedclustersetbinding is created in openshift["project"]
+* gateway class "kuadrant-multi-cluster-gateway-instance-per-cluster" is created
+* Let's Encrypt ClusterIssuer object configured on the cluster
+
+"""
+
+import dataclasses
+from importlib import resources
+
+import pytest
+
+from testsuite.gateway import Exposer, CustomReference
+from testsuite.gateway.gateway_api.hostname import DNSPolicyExposer
+
+pytestmark = [pytest.mark.mgc]
+
+
+@pytest.fixture(scope="module")
+def cluster_issuer():
+    """Reference to cluster Let's Encrypt certificate issuer"""
+    return CustomReference(
+        group="cert-manager.io",
+        kind="ClusterIssuer",
+        name="letsencrypt-staging",
+    )
+
+
+@pytest.fixture(scope="module")
+def exposer(base_domain, hub_gateway) -> Exposer:
+    """DNSPolicyExposer setup with expected TLS certificate"""
+    external_cert = resources.files("testsuite.resources").joinpath("letsencrypt-stg-root-x1.pem").read_text()
+    old_cert = hub_gateway.get_tls_cert()
+    # Removed irrelevant intermediate certificate used for backwards compatibility
+    trimmed_cert = old_cert.certificate.rsplit("-----BEGIN CERTIFICATE-----", 1)[0]
```

Comment:
@pehala I looked into why using the entire cert chain to verify wasn't working and it turns the cert gets generated with 2 intermediates connected to it, one of them leading to a root that's long expired (for backwards compatibility) so I removed that one from the chain. This may introduce problems one day if the default chain gets switched to the blue path[ (link)](https://letsencrypt.org/2020/12/21/extending-android-compatibility.html)

Alternative solution would be to force our let's encrypt cluster issuer to use the alternative chain [(link)](https://cert-manager.io/docs/configuration/acme/#alternative-certificate-chains)

Reply:
After a further discussion we decided to use the alternative chain by default, an example on how to configure the ClusterIssuer to use the alternative chain was added to the test description